enaml/core/dynamicscope.py

```diff
@@ -0,0 +1,58 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2025, Nucleic Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ------------------------------------------------------------------------------
+from itertools import chain, repeat
+from enaml.core._dynamicscope import _DynamicScope, UserKeyError
+
+
+class DynamicScope(_DynamicScope):
+    """_DynamicScope is a C++ class which exposes the following attributes:
+
+    _owner
+    _change
+    _f_writes
+    _f_locals
+    _f_globals
+    _f_builtins
+
+    """
+
+    def __iter__(self):
+        fwrites_it = iter(self._f_writes or ())
+        self_it = repeat("self", 1)
+        change_it = repeat("change", 1 if self._change else 0)
+        flocals_it = iter(self._f_locals)
+        fglobals_it = iter(self._f_globals)
+        fbuiltins_it = iter(self._f_builtins)
+        fields = set()
+        parent = self._owner
+        while parent is not None:
+            fields.update(dir(parent))
+            parent = parent._parent
+        parent_it = iter(fields)
+        return filter(lambda s: not s.startswith("__"), chain(
+            fwrites_it,
+            self_it,
+            change_it,
+            flocals_it,
+            fglobals_it,
+            fbuiltins_it,
+            parent_it,
+        ))
+
+    def keys(self):
+        return iter(self)
+
+    def values(self):
+        return tuple(self[key] for key in self)
```

Comment:
can we not return a generator expression for this method and `items`, or is it only `keys` that can return an iterator?

Reply:
Good point, not sure why I changed that.